src/logic/safe/store/actions/fetchSafe.ts

```diff
@@ -58,6 +60,7 @@ export const buildSafe = async (safeAddress: string): Promise<SafeRecordProps> =
 export const fetchSafe =
   (safeAddress: string, isInitialLoad = false) =>
   async (dispatch: Dispatch<any>): Promise<Action<Partial<SafeRecordProps>> | void> => {
+    dispatch(addNftTokens(NFTTokensDefaultState))
```

Comment:
The loaded state for NFT tokens didn't reset when switching safes. Not sure if this is the best way to handle resetting the state. @iamacook @katspaugh wdyt?

Reply:
The loaded flag for NFTs was somewhat of a workaround. Can we use your newly included flag in unison?